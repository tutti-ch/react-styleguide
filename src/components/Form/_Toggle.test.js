/* global describe, test, jest, expect */
import React from "react";
import Toggle, { Toggle as TogglePure } from "./_Toggle";
import { shallow, mount } from "enzyme";

describe("(Component) Toggle", function() {
  test("[handleOnClick] should get the attribute and update the state", () => {
    const onChange = jest.fn();
    const comp = shallow(<TogglePure onChange={onChange} name="toggle" />);
    const inst = comp.instance();
    const event = {
      preventDefault: jest.fn()
    };

    const spy = jest.spyOn(inst, "setState");
    inst.handleOnClick(event, "Value1");
    expect(event.preventDefault).toHaveBeenCalled();
    expect(spy).toHaveBeenCalledWith({ value: "Value1" }, expect.any(Function));
    expect(onChange).toHaveBeenCalledWith("Value1", {
      name: "toggle",
      initialValue: "",
      formValue: "Value1"
    });

    // Should not work when disabled
    comp.setProps({ disabled: true });
    inst.handleOnClick(event);
    expect(event.preventDefault).toHaveBeenCalledTimes(1);
  });

  test("[componentDidUpdate] should update the state when the value changes", () => {
    const onChange = jest.fn();
    const comp = shallow(<TogglePure onChange={onChange} name="toggle" />);
    const inst = comp.instance();
    const spy = jest.spyOn(inst, "setState");
    comp.setProps({ value: "My value" });
    expect(spy).toHaveBeenCalled();
  });

  test("should match snapshots", () => {
    const comp = mount(
      <Toggle value="val1">
        <Toggle.Option value="val1">Value 1</Toggle.Option>
        <Toggle.Option value="val2">Value 2</Toggle.Option>
      </Toggle>
    );
    expect(comp).toMatchSnapshot();
  });
});
