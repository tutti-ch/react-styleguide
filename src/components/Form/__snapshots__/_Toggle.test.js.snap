// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`(Component) Toggle should match snapshots 1`] = `
<WithWrapper
  value="val1"
>
  <div
    className=""
  >
    <Toggle
      onChange={[Function]}
      onKeyUp={[Function]}
      value="val1"
    >
      <span>
        <span
          className=""
        >
          <Component
            key=".0"
            onClick={[Function]}
            selected={true}
            value="val1"
          >
            <span
              className="undefined"
              data-value="val1"
              onClick={[Function]}
            >
              Value 1
            </span>
          </Component>
          <Component
            key=".1"
            onClick={[Function]}
            selected={false}
            value="val2"
          >
            <span
              className=""
              data-value="val2"
              onClick={[Function]}
            >
              Value 2
            </span>
          </Component>
        </span>
        <input
          type="hidden"
          value="val1"
        />
      </span>
    </Toggle>
  </div>
</WithWrapper>
`;
